### ISSUE TITLE
drt: consider having long-running sessions with some simple workload

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#122155]
In order for us to reproduce and catch some bugs (e.g. #121844), it might require to have sessions that are running for weeks and only close when the nodes restart. We cannot really replicate such scenario neither in CI nor in roachtests, and the DRT cluster seems like a perfect fit. We should consider introducing a simple workload that would run continuously / periodically on such long-running-session.

Jira issue: CRDB-37733

**Comment by yuzefovich** [comment]
[triage] does tpcc workload already do this for us? do we run it long enough on DRT cluster? Should DRT team look into this overall?

[michae2] concerned that spot instances that would be going down frequently would be killing long-living sessions preventing implementing a workload like this issue describes.

cc @BabuSrithar @srosenberg

**Comment by srosenberg** [comment]
> We observed on one of the CC clusters that a session that issued 3M txns has "memory usage" reported as around 400MiB.

For the regression test, would it suffice to _assert_ on the memory usage of each long-lived session? Using `crdb_internal.node_memory_monitors` for observability?

**Comment by yuzefovich** [comment]
For regression test for that particular bug, yes, we could do that. This issue is more general - about having very long-lived sessions since some of our customers never close connections (unless the nodes are restarted).

**Comment by srosenberg** [comment]
> For regression test for that particular bug, yes, we could do that. This issue is more general - about having very long-lived sessions in our tests _somewhere_ since some of our customers never close connections (unless the nodes are restarted).

Yep, it makes sense! I was just confirming the general test strategy. This is a great candidate for long-running clusters. What about perturbations? A long-running cluster will inevitably experience external (and internal) failures; this would make it trickier to keep persistent sessions. (I assume a disconnect would be a deal breaker; i.e., would it resolve the memory leak in this case?)

**Comment by yuzefovich** [comment]
Totally, I don't expect any kind of guarantee on the lifetime of these sessions, rather it'd be a good addition to our test suite to have sessions that are as long-lived as possible, on a best-effort basis. Any node restart or network hiccups should be ignored. My hope is that even best-effort with no guarantees would be sufficient to tickle some existing bugs / prevent new regressions.

**Comment by rytaft** [comment]
Not sure why this was added back to SQL Queries. Removing it from our board since it looks like the DRP team is working on it.

**Comment by csgourav** [comment]
cockroach workload has `--max-conn-lifetime=1h` flag which can be used to increase the connection pool active time, by default it is 5m. This flag governs the session active duration time. We can adjust this time for all workloads running on drt-large.

@rytaft What would be the ideal time you want for the session lifetime, should we increase it to like 12h or you want more than that like 24h or 72h?

**Comment by rytaft** [comment]
Hi @csgourav -- the issue description says "it might require to have sessions that are running for weeks and only close when the nodes restart." So ideally we'd want the sessions to be running as long as possible. Thank you!

**Comment by rytaft** [comment]
> @rytaft , we can change the workload configuration with --max-conn-lifetime=168h, which is 1 week.

@nameisbhaskar that sounds good, although as noted in [the issue description](https://github.com/cockroachdb/cockroach/issues/122155#issue-2236405901), testing for connections lasting multiple weeks would be ideal. But one week is a great start!

**Comment by dev-kyle** [comment]
Picking this up!

Planning to add `MAX_CONN_LIFETIME: 720h` (30 days) to the `drt_large.yaml`. The env var is already checked in `tpcc_run_multiregion.sh`, so should just be a one-line change (will need to do more assessing to see what else will need to be modified accordingly). One month seems like a good starting point for the ideal "multiple weeks".

Scoping to `drt-large` only, let me know if any other DRT clusters should get the same treatment.

**Comment by dev-kyle** [comment]
Quick update: `drt-large` doesn't appear to be actively running, and `drt_large.yaml` hasn't had any substantial changes in a while (last real activity Jan 2025).

It looks like (according to `drtprod list`) `drt-1m-tables` and `drt-upgrade-test` are the only running clusters currently. Are these the only two candidates currently to introduce long-running sessions, or will other clusters be revived that this can be attached to as well?
